app/javascript/components/cloud-networks/nsxt-cloud-network-form.jsx

```diff
@@ -0,0 +1,65 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import MiqFormRenderer from '@@ddf';
+import createSchema from './nsxt-cloud-network-form.schema';
+import { CloudNetworkApi } from '../../utils/cloud-network-api'
+import { ProvidersApi } from '../../utils/providers.api';
+
+const NsxtCloudNetworkForm = ({ recordId }) => {
+  const [{ isLoading, initialValues }, setState] = useState({
+    isLoading: !!recordId,
+    initialValues: {},
+  });
+
+  useEffect(() => {
+    if (recordId) {
+      API.get(`/api/cloud_networks/${recordId}`).then(({ id, providerId, ...initialValues }) => {
+        setState({
+          isLoading: false,
+          initialValues,
+        });
+      });
+    }
+  }, [recordId]);
+
+  const initialize = (formOptions) => {
+    ManageIQ.redux.store.dispatch({
+      type: 'FormButtons.init',
+      payload: {
+        newRecord: true,
+        pristine: true,
+      },
+    });
+
+    ManageIQ.redux.store.dispatch({
+      type: 'FormButtons.callbacks',
+      payload: { addClicked: () => formOptions.submit() },
+    });
+  };
+
+  const onSubmit = (values) => {
+    miqSparkleOn();
+    const request = recordId ? API.patch(`/api/cloud_networks/${recordId}`, values) : API.post('/api/cloud_networks', values);
```

Comment:
We need to update the API to allow this patch/post API call.

This is working with the current API.
```
    const request = !!recordId && ManageIQ.controller === 'cloud_network' ? 
      API.post(`/api/providers/${values.ems_id}/cloud_networks`, {
        action: 'edit',
        id: values.id,
        resource: { ...values },
      }) : 
      API.post(`/api/providers/${values.ems_id}/cloud_networks`, {
        action: 'create',
        resource: values,
      });
```